src/main/java/me/squidxtv/frameui/api/builder/ScreenBuilder.java

```diff
@@ -0,0 +1,99 @@
+package me.squidxtv.frameui.api.builder;
+
+import me.squidxtv.frameui.core.Renderer;
+import me.squidxtv.frameui.core.Screen;
+import me.squidxtv.frameui.math.Direction;
+import org.bukkit.Location;
+import org.bukkit.entity.Player;
+
+/**
+ * Builder class for creating and configuring instances of {@link Screen}.
+ * This builder uses a fluent API style, enabling method chaining for ease of use.
+ *
+ * <p>Example usage:</p>
+ *
+ * <pre>
+ * {@code
+ * public void example(ScreenRegistry registry) {
+ *     var player = Bukkit.getPlayer("mike");
+ *     var screen = registry.create(this)
+ *             .withSize(10, 10)
+ *             .withViewer(player)
+ *             .withName("Main Menu Screen")
+ *             .withLocation(player.getLocation())
+ *             .build();
+ *
+ *     screen.open();
+ * }
+ * }
+ * </pre>
+ */
+public interface ScreenBuilder {
+
+    /**
+     * @param width the number of horizontal item frames
+     */
```

Comment:
Same here and the other changes, adding a general javadoc comment for consistency

Reply:
done